lib/purescript-ast/src/Language/PureScript/Environment.hs

```diff
@@ -377,6 +377,13 @@ isTypeOrApplied :: Type a -> Type b -> Bool
 isTypeOrApplied t1 (TypeApp _ t2 _) = eqType t1 t2
 isTypeOrApplied t1 t2 = eqType t1 t2
 
+kindArity :: SourceType -> Int
+kindArity = go' 0 where
+  go' n (TypeApp _ (TypeApp _ k _) b)
+    | k == tyFunction = go' (n + 1) b
+  go' n (ForAll _ _ _ k _) = go' n k
+  go' n _ = n
```

Comment:
I would’ve liked to export this function from `Language.PureScript.Types` instead but there would then be a circular dependency because `kindArity` depends on `Language.PureScript.Environment.tyFunction` and `Language.PureScript.Environment` depends on `Language.PureScript.Types`.